File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var minSubsequence = function(nums) {
  if (nums.length < 2) return nums
  const sum = nums.reduce((a, b) => a+b)
  nums.sort((a, b) => b - a)
  const ans = []
  let rest = sum / 2
  for (let i = 0; i < nums.length && rest >= 0; i++) {
    const n = nums[i]
    ans.push(n)
    rest -= n
  }
  return ans
};

File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.go
func minSubsequence(nums []int) []int {
  sort.Ints(nums)
  sum := 0
  for _, v := range nums {
    sum += v
  }
  ans := make([]int, 0, len(nums))
  max := 0
  for i := len(nums) - 1; i >= 0; i-- {
    ans = append(ans, nums[i])
    max += nums[i]
    if (max<<1) > sum {
      return ans
    }
  }
  return []int{}
}